taichi/program/state_flow_graph.h

```diff
@@ -28,14 +28,42 @@ class StateFlowGraph {
     std::string task_name;
     // Incremental ID to identify the i-th launch of the task.
     int launch_id;
+    bool is_initial_node{false};
+
+    std::unordered_set<AsyncState, AsyncStateHash> input_states, output_states;
```

Comment:
TODO: In the future, maybe we can just store a pointer/reference to `TaskMeta`..